src/trusted/service_runtime/nacl_syscall_common.c

```diff
@@ -578,68 +620,93 @@ int32_t NaClSysOpen(struct NaClAppThread  *natp,
   retval = NaClHostDescStat(path, &stbuf);
 
   /* Windows does not have S_ISDIR(m) macro */
-  if (0 == retval && S_IFDIR == (S_IFDIR & stbuf.st_mode)) {
+  if (!retval && S_IFDIR == (S_IFDIR & stbuf.st_mode)) {
     struct NaClHostDir  *hd;
 
     hd = malloc(sizeof *hd);
-    if (NULL == hd) {
+    if (!hd) {
       retval = -NACL_ABI_ENOMEM;
       goto cleanup;
     }
     retval = NaClHostDirOpen(hd, path);
     NaClLog(1, "NaClHostDirOpen(0x%08"NACL_PRIxPTR", %s) returned %d\n",
             (uintptr_t) hd, path, retval);
-    if (0 == retval) {
-      retval = NaClSetAvail(nap,
-                            ((struct NaClDesc *) NaClDescDirDescMake(hd)));
-      NaClLog(1, "Entered directory into open file table at %d\n",
-              retval);
+    if (!retval) {
+      retval = NaClSetAvail(nap, ((struct NaClDesc *) NaClDescDirDescMake(hd)));
+      NaClLog(1, "Entered directory into open file table at %d\n", retval);
     }
   } else {
     struct NaClHostDesc  *hd;
 
     hd = malloc(sizeof *hd);
-    if (NULL == hd) {
+    if (!hd) {
       retval = -NACL_ABI_ENOMEM;
       goto cleanup;
     }
     retval = NaClHostDescOpen(hd, path, flags, mode);
-    NaClLog(1,
-            "NaClHostDescOpen(0x%08"NACL_PRIxPTR", %s, 0%o, 0%o) returned %d\n",
+    NaClLog(1, "NaClHostDescOpen(0x%08"NACL_PRIxPTR", %s, 0%o, 0%o) returned %d\n",
             (uintptr_t) hd, path, flags, mode, retval);
-    if (0 == retval) {
-      retval = NaClSetAvail(nap,
-                            ((struct NaClDesc *) NaClDescIoDescMake(hd)));
+    if (!retval) {
+      retval = NaClSetAvail(nap, ((struct NaClDesc *) NaClDescIoDescMake(hd)));
       NaClLog(1, "Entered into open file table at %d\n", retval);
     }
   }
+
 cleanup:
-  return retval;
+  /* yiwen: now translate the real fds to virtual fds and return them to the cage */
+  fd_cage_table[nap->cage_id][nap->fd] = retval;
+  fd_retval = nap->fd;
+  nap->fd++;
+
+  /*
+   * register the fd and lib_path info for the cage, in lib_table[CACHED_LIB_NUM_MAX]
+   * this will be used when trying to check if a lib has been cached in our system
+   */
+  if (fd_retval > CACHED_LIB_NUM_MAX) {
+     strncpy(nap->lib_table[fd_retval].path, path, strlen(path) + 1);
+     nap->num_lib++;
+  }
+
+  NaClLog(1, "[*** Debug ***][Open] fd = %d, filepath = %s \n", fd_retval, path);
```

Comment:
Besides the hardcoded log level mentioned previously, the debug message is quite different from the others I saw so far: custom prefix, contains only half of the function name; trailing space in message.

Reply:
Whoops, missed that. Will fix.